addons/hr_attendance/models/hr_attendance.py

```diff
@@ -220,7 +215,8 @@ def _get_day_start_and_day(self, employee, dt):
         #Returns a tuple containing the datetime in naive UTC of the employee's start of the day
         # and the date it was for that employee
         if not dt.tzinfo:
-            date_employee_tz = pytz.utc.localize(dt).astimezone(pytz.timezone(employee._get_tz()))
+            date_from = dt.replace(hour=0, minute=0, second=0) - timedelta(days=1)
```

Comment:
Not sure to understand what we do here, but as we broaden the window in the past, I think it could only fix the issue in a single direction (from USA to Japan for instance) and not on the other one. Could you confirm ?

Reply:
Here the date is actually "only" used [to find matching contracts](https://github.com/odoo/odoo/blob/dcf0880f836423ee82f3d283ffbf940a804bc13b/addons/hr_contract/models/hr_employee.py#L120-L130) to get a tz from; so I don't think it would cause the problem you're thinking of.
I removed a day to avoid having more timezones issues but looking at the query again we also filter for both date_end and date_start so in fact this doesn't work at all and we have a problem because strictly speaking this should be the very thing we're trying to compute... 
The problem could happen for attendances on the last or first day of a contract.